### PR TITLE
docs: Mention how to build images for local CI testing

### PR DESCRIPTION
**Pull request by brb**

### CODE REVIEW

File: Documentation/contributing/testing/e2e.rst [hunk 1]
```diff
@@ -287,6 +287,30 @@ needing to run the full test, you can build the test directory:
...
+
+   cd go/src/github.com/cilium/cilium
+
+   make LOCKDEBUG=1 docker-cilium-image
+   docker tag quay.io/cilium/cilium:latest k8s1:5000/cilium/cilium-dev
+   docker push k8s1:5000/cilium/cilium-dev
+
+   make -B LOCKDEBUG=1 docker-operator-generic-image
+   docker tag quay.io/cilium/operator-generic:latest
+   k8s1:5000/cilium/operator-generic:latest
```
Comment (qmonnet) [inline Documentation/contributing/testing/e2e.rst]:
```suggestion
   docker tag quay.io/cilium/operator-generic:latest \
           k8s1:5000/cilium/operator-generic:latest
```

File: Documentation/contributing/testing/e2e.rst [hunk 2]
```diff
@@ -287,6 +287,30 @@ needing to run the full test, you can build the test directory:
...
+
+.. code-block:: shell-session
+
+   cd go/src/github.com/cilium/cilium
+
+   make LOCKDEBUG=1 docker-cilium-image
+   docker tag quay.io/cilium/cilium:latest k8s1:5000/cilium/cilium-dev
+   docker push k8s1:5000/cilium/cilium-dev
+
+   make -B LOCKDEBUG=1 docker-operator-generic-image
```
Comment (qmonnet) [inline Documentation/contributing/testing/e2e.rst:308]: Do you know if/why the `-B` is necessary here?

File: Documentation/contributing/testing/e2e.rst [hunk 3]
```diff
@@ -287,6 +287,30 @@ needing to run the full test, you can build the test directory:
...
+built, and then used by the CI suite. To do so, one can run the following
+commands on the ``k8s1`` VM:
+
+.. code-block:: shell-session
+
+   cd go/src/github.com/cilium/cilium
+
+   make LOCKDEBUG=1 docker-cilium-image
+   docker tag quay.io/cilium/cilium:latest k8s1:5000/cilium/cilium-dev
+   docker push k8s1:5000/cilium/cilium-dev
```
Comment (qmonnet) [inline Documentation/contributing/testing/e2e.rst]: Should this image be tagged as `latest` too?

```suggestion
   docker tag quay.io/cilium/cilium:latest k8s1:5000/cilium/cilium-dev:latest
   docker push k8s1:5000/cilium/cilium-dev:latest
```

**Review by qmonnet — CHANGES_REQUESTED** [review]
There's a `docker tag` command that got split and needs to be fixed, please see suggestion below.

Other than that, and looking at the “template” in `Makefile.docker`, I believe most of those can be replaced with a single command? Something like this:

```
$ cd go/src/github.com/cilium/cilium

$ DOCKER_REGISTRY=k8s1:5000 ARCH=amd64 DOCKER_FLAGS=--push LOCKDEBUG=1 \
           make dev-docker-image docker-operator-generic-image
```

**Comment by brb** [comment]
> DOCKER_REGISTRY=k8s1:5000 ARCH=amd64 DOCKER_FLAGS=--push LOCKDEBUG=1 \
           make dev-docker-image docker-operator-generic-image

Great suggestion! However, it seems to fail with and without `--push`:
```
 => => pushing layers                                                                                                                                                                                            0.0s
------
 > exporting to image:
------
error: failed to solve: failed to do request: Head "https://k8s1:5000/v2/cilium/cilium-dev/blobs/sha256:2ff0e44daf075434788008b724601efcdd10bbe7c3d2a292fe8e82312fe762e1": dial tcp: lookup k8s1 on 8.8.8.8:53: no such host
Makefile.docker:107: recipe for target 'dev-docker-image' failed
make: *** [dev-docker-image] Error 1
```

I guess k8s1 (defined in /etc/hosts) is not available in the buildx context.

File: Documentation/contributing/testing/e2e.rst (hunk 2 above)
Comment (brb) [reply Documentation/contributing/testing/e2e.rst:308]: No idea (blindly taken from compile.sh).

File: Documentation/contributing/testing/e2e.rst (hunk 3 above)
Comment (brb) [reply Documentation/contributing/testing/e2e.rst]: Yep (although not specifying it defaults to `latest`).

**Review by qmonnet — APPROVED** [review]
Ok, sorry for the misleading suggestion. Let's keep your version then.
Looks good to me!

**Comment by aditighag** [comment]
The instructions don't work for me - 

```
$ REGISTRY=k8s1:5000 ARCH=amd64 DOCKER_FLAGS=--push LOCKDEBUG=1            make dev-docker-image
make: Warning: File 'GIT_VERSION' has modification time 5121 s in the future
Using Docker Buildx builder "cool_ellis" with build flags "--push --push --platform linux/amd64".
docker buildx build -f images/cilium/Dockerfile --push --push --platform linux/amd64 --build-arg BASE_IMAGE=scratch --build-arg NOSTRIP= --build-arg NOOPT= --build-arg LOCKDEBUG=1 --build-arg RACE= --build-arg V= --build-arg LIBNETWORK_PLUGIN= --build-arg CILIUM_SHA=734848e230 --build-arg OPERATOR_VARIANT=cilium-dev -t quay.io/cilium/cilium-dev:latest .
[+] Building 15.0s (1/1) FINISHED                                                                                                                                                                                                                     
 => ERROR [internal] booting buildkit                                                                                                                                                                                                           15.0s
 => => pulling image moby/buildkit:buildx-stable-1                                                                                                                                                                                              15.0s
------
 > [internal] booting buildkit:
------
error: Error response from daemon: Get "https://registry-1.docker.io/v2/": context deadline exceeded
Makefile.docker:108: recipe for target 'dev-docker-image' failed

```

It's not clear to me why it's trying to reach the external URL if we are using a local registry. Haven't had time to dig into the issue yet.

**Comment by brb** [comment]
@aditighag Please see the instructions in the PR.

**Review by aditighag — APPROVED** [review]
There was a problem on my local test cluster. The commands that were failing earlier are now working.

**Comment by qmonnet** [comment]
Travis failure is due to Docker rate-limiting the image pulls, this is safe to ignore.

**Pull request merged**